statsmodels/gam/tests/test_penalized.py

```diff
@@ -0,0 +1,571 @@
+        cls.pw = mod.pen_weight = 1 / s_scale_r / 2
+        cls.res1 = mod.fit(cov_type=cls.cov_type, method='bfgs', maxiter=100,
+                           disp=0, trim=False, scale='x2')
+        cls.res2 = results_pls.pls5
+
+        cls.rtol_fitted = 1e-5
+        # edf is currently not available with PenalizedMixin
+        # need correction for difference in scale denominator
+        cls.covp_corrfact = 1.0025464444310588
+
+    def _test_cov_robust(self):
+        # TODO: HC0 differs from Theil sandwich, difference is large
+        res1 = self.res1
+        res2 = self.res2
+        pw = res1.model.pen_weight
+        res1 = res1.model.fit(pen_weight=pw, cov_type='HC0')
+        assert_allclose(np.asarray(res1.cov_params()),
+                        res2.Ve * self.covp_corrfact, rtol=1e-4)
+
+
+class TestGAM5Pirls(CheckGAMMixin):
+
+    cov_type = 'nonrobust'
+
+    @classmethod
+    def setup_class(cls):
+        s_scale = 0.0263073404164214
+
+        x = data_mcycle['times'].values
+        endog = data_mcycle['accel']
+        cc = CyclicCubicSplines(x, df=[6], constraints='center')
+        gam_cc = GLMGam(endog, smoother=cc, alpha=1 / s_scale / 2)
+        cls.res1 = gam_cc.fit()
+        cls.res2 = results_pls.pls5
+
+        cls.rtol_fitted = 1e-12
+        # cls.covp_corrfact = 1.0025464444310588  # without edf
+        # edf is implemented
+        cls.covp_corrfact = 1
+
+
+class TestGAM5Bfgs(CheckGAMMixin):
+
+    cov_type = 'nonrobust'
+
+    @classmethod
+    def setup_class(cls):
+        s_scale = 0.0263073404164214
+
+        x = data_mcycle['times'].values
+        endog = data_mcycle['accel']
+        cc = CyclicCubicSplines(x, df=[6], constraints='center')
+        gam_cc = GLMGam(endog, smoother=cc, alpha=1 / s_scale / 2)
+        cls.res1 = gam_cc.fit(method='bfgs')
+        cls.res2 = results_pls.pls5
+
+        cls.rtol_fitted = 1e-5
+        # cls.covp_corrfact = 1.0025464444310588  # without edf
+        # edf is implemented
+        cls.covp_corrfact = 1
+
+    def test_predict(self):
+        res1 = self.res1
+        res2 = self.res2
+        predicted = res1.predict(None, res1.model.smoother.x[2:4])
+        assert_allclose(predicted, res1.fittedvalues[2:4],
+                        rtol=1e-13)
+        assert_allclose(predicted, res2.fitted_values[2:4],
+                        rtol=self.rtol_fitted)
+
+
+class TestGAM6Pirls(object):
+
+    @classmethod
+    def setup_class(cls):
+        s_scale = 0.0263073404164214
+
+        cc = CyclicCubicSplines(data_mcycle['times'].values, df=[6])
+        gam_cc = GLMGam(data_mcycle['accel'], smoother=cc,
+                        alpha=1 / s_scale / 2)
+        cls.res1 = gam_cc.fit()
+
+    def test_fitted(self):
+        res1 = self.res1
+        pred = res1.get_prediction()
+        self.rtol_fitted = 1e-7
+        pls6_fittedvalues = np.array([
+            2.45008146537851, 3.14145063965465, 5.24130119353225,
```

Comment:
All the results are from R mgcv, unless regression test is explicitly mentioned.
These are cases, where I just wanted to test a few numbers from an interactive R session instead of putting all those numbers in csv files.
I'm not sure whether I have a log of the corresponding R commands.